account_financial_report/report/trial_balance.py

```diff
@@ -0,0 +1,577 @@
+    coalesce(rag.final_debit - rag.initial_debit, 0) AS debit,
+    coalesce(rag.final_credit - rag.initial_credit, 0) AS credit,
+    coalesce(rag.final_balance - rag.initial_balance, 0) AS period_balance,
+    coalesce(rag.final_balance, 0) AS final_balance,
+    rag.currency_id AS currency_id,
+    coalesce(rag.initial_balance_foreign_currency, 0)
+        AS initial_balance_foreign_currency,
+    coalesce(rag.final_balance_foreign_currency, 0)
+        AS final_balance_foreign_currency
+FROM
+    account_account acc
+    LEFT OUTER JOIN report_general_ledger_account AS rag
+        ON rag.account_id = acc.id AND rag.report_id = %s
+WHERE
+    acc.id in %s
+        """
+        query_inject_account_params = (
+            self.id,
+            self.env.uid,
+            self.general_ledger_id.id,
+            account_ids._ids,
+        )
+        self.env.cr.execute(query_inject_account, query_inject_account_params)
+
+    def _inject_partner_values(self):
+        """Inject report values for report_trial_balance_partner"""
+        query_inject_partner = """
+INSERT INTO
+    report_trial_balance_partner
+    (
+    report_account_id,
+    create_uid,
+    create_date,
+    partner_id,
+    name,
+    initial_balance,
+    initial_balance_foreign_currency,
+    debit,
+    credit,
+    period_balance,
+    final_balance,
+    final_balance_foreign_currency
+    )
+SELECT
+    ra.id AS report_account_id,
+    %s AS create_uid,
+    NOW() AS create_date,
+    rpg.partner_id,
+    rpg.name,
+    rpg.initial_balance AS initial_balance,
+    rpg.initial_balance_foreign_currency AS initial_balance_foreign_currency,
+    rpg.final_debit - rpg.initial_debit AS debit,
+    rpg.final_credit - rpg.initial_credit AS credit,
+    rpg.final_balance - rpg.initial_balance AS period_balance,
+    rpg.final_balance AS final_balance,
+    rpg.final_balance_foreign_currency AS final_balance_foreign_currency
+FROM
+    report_general_ledger_partner rpg
+INNER JOIN
+    report_general_ledger_account rag ON rpg.report_account_id = rag.id
+INNER JOIN
+    report_trial_balance_account ra ON rag.code = ra.code
+WHERE
+    rag.report_id = %s
+AND ra.report_id = %s
+        """
+        query_inject_partner_params = (
+            self.env.uid,
+            self.general_ledger_id.id,
+            self.id,
+        )
+        self.env.cr.execute(query_inject_partner, query_inject_partner_params)
+
+    def _inject_account_group_values(self):
+        """Inject report values for report_trial_balance_account"""
+        query_inject_account_group = """
+INSERT INTO
+    report_trial_balance_account
+    (
+    report_id,
+    create_uid,
+    create_date,
+    account_group_id,
+    parent_id,
+    code,
+    name,
+    sequence,
+    level
+    )
+SELECT
+    %s AS report_id,
+    %s AS create_uid,
+    NOW() AS create_date,
+    accgroup.id,
+    accgroup.parent_id,
+    coalesce(accgroup.code_prefix, accgroup.name),
+    accgroup.name,
+    accgroup.id * 100000,
```

Comment:
I think there is an issue here. Account group ordering is based on database id, and not the hierarchy of account groups

Reply:
This is for having unique IDs, so it's correct, unless you have more than 100k accounts.

Reply:
No, this is for the sequence field of report_trial_balance_account. Check line 393

Reply:
OK, I see, but this is working in previous versions, so if this hasn't been changed, then it should work through other logic. I have to say that I hate all this SQL stuff, but it's what we have for now, and If you are only checking code and don't have specific issue trying the module, then it's not a bug.

Reply:
previous version used parent_left which was okay because the field is sorted based on the hierarchy. Version 12 doesn't have the concept of parent_left/parent_right, and using parent_id won't help either.

The only mitigation I can think of is to depend on the code_prefix for ordering. Not the best option, but for sure much better than the account database id.

Reply:
https://github.com/OCA/account-financial-reporting/blob/11.0/account_financial_report/report/trial_balance.py#L404

Reply:
OK, then this has been changed here... Maybe the red CIs is due to these changes. Any way, you still have `parent_path` field for getting ordered records. Why not use it?

I plan to rewrite this for avoiding the SQLs in the future for finally remove this SQL hell, but not sure when.

Reply:
@osoul @pedrobaeza should this be `accgroup.parent_id * 100000`?

Reply:
Considering that in v11 is `accgroup.parent_left * 100000`

Reply:
Now, that doesn't serve, as the parent_id database IDs are also not ordered. What we can try is to use `parent_path` + account name and ordering that way.

Reply:
@pedrobaeza thanks, see 218a0a0149e4c671a65b5763fa3374a4ecb0e74e